src/services/activeUsers.js

```diff
@@ -0,0 +1,124 @@
+import { Query } from 'pg';
+import through2 from 'through2';
+
+import {
+  sequelize,
+} from '../models';
+
+const arrayFields = [
+  'Roles',
+  'READ_WRITE_REPORTS regions',
+  'READ_REPORTS regions',
+  'APPROVE_REPORTS regions',
+  'UNLOCK_APPROVED_REPORTS regions',
+  'Flags',
+];
+
+/**
+ * Transform function to convert the data to the right format.
+ *
+ * @param {*} user - a user row
+ */
+function convertToCSV(user) {
+  // Surround array fields with double quotes
+  const mappedUser = {};
+  Object.keys(user)
+    .forEach((k) => { mappedUser[k] = (arrayFields.includes(k) && user[k] ? `"${user[k]}"` : user[k]); });
+
+  const result = Object.values(mappedUser).toString();
+
+  return `${result}\n`;
+}
+
+/**
+ * Transform function to convert the data to the right format.
+ *
+ * @param {*} sql - raw sql
+ * @param {*} transform - a function to transform a row of data
+ * @returns {*} - a new stream
+ */
+const streamable = async (sql, transform) => {
+  const conn = await sequelize.connectionManager.getConnection();
+  const stream = through2.obj();
+  const query = conn.query(new Query(sql));
+
+  let writeHeader = true;
+
+  const end = async () => {
+    await sequelize.connectionManager.releaseConnection(conn);
+    stream.end();
+  };
+  query.on('row', (r) => {
+    if (writeHeader) {
+      writeHeader = false;
+      const header = Object.keys(r).toString();
+      stream.push(`${header}\n`);
+    }
+    stream.push(transform ? transform(r) : r);
+  });
+  query.once('error', (err) => {
+    stream.emit('error', err);
+    end();
+  });
+  query.once('end', end);
+
+  return stream;
+};
```

Comment:
Is this needed? the result set is less then 600 rows and the columns are small. why not just load all the data at once and process as normal. the "group by"'s are not too complicated  that this could have been done within a normal sequelize model findAll.

Reply:
I was wrong findAll will not work as the filter expressions are not supported yet https://github.com/sequelize/sequelize/issues/5732